subprojects/gradle/test-summary/src/main/kotlin/com/avito/test/summary/TestSummaryPlugin.kt

```diff
@@ -1,115 +1,52 @@
 package com.avito.test.summary
 
-import com.avito.android.stats.statsd
-import com.avito.http.HttpClientProvider
+import com.avito.android.Problem
+import com.avito.android.asPlainText
+import com.avito.kotlin.dsl.isRoot
 import com.avito.logger.GradleLoggerFactory
-import com.avito.logger.LoggerFactory
-import com.avito.report.ReportsApi
-import com.avito.report.ReportsApiFactory
-import com.avito.report.model.Team
-import com.avito.slack.SlackClient
+import com.avito.logger.create
 import com.avito.time.DefaultTimeProvider
 import com.avito.time.TimeProvider
 import org.gradle.api.Plugin
 import org.gradle.api.Project
-import org.gradle.api.provider.Provider
 import org.gradle.kotlin.dsl.create
 import org.gradle.kotlin.dsl.register
 
-class TestSummaryPlugin : Plugin<Project> {
+public class TestSummaryPlugin : Plugin<Project> {
 
     override fun apply(target: Project) {
 
-        val extension = target.extensions.create<TestSummaryExtension>(testSummaryExtensionName)
-
         val loggerFactory = GradleLoggerFactory.fromPlugin(this, target)
 
-        val timeProvider: TimeProvider = DefaultTimeProvider()
-
-        @Suppress("UnstableApiUsage")
-        val slackClient: Provider<SlackClient> =
-            extension.slackToken.zip(extension.slackWorkspace) { token, workspace ->
-                createSlackClient(
-                    slackToken = token,
-                    slackWorkspace = workspace,
-                    httpClientProvider = HttpClientProvider(
-                        statsDSender = target.statsd.get(),
-                        timeProvider = timeProvider,
-                        loggerFactory = loggerFactory
-                    )
-                )
-            }
+        val logger = loggerFactory.create<TestSummaryPlugin>()
 
-        val reportsApi: Provider<ReportsApi> = extension.reportsHost.map {
-            createReportsApi(
-                reportsHost = it,
-                loggerFactory = loggerFactory,
-                httpClientProvider = HttpClientProvider(
-                    statsDSender = target.statsd.get(),
-                    timeProvider = timeProvider,
-                    loggerFactory = loggerFactory
-                )
+        if (!target.isRoot()) {
+            val problem = Problem(
+                shortDescription = "TestSummaryPlugin should be applied to root project",
+                context = "TestSummaryPlugin applied to ${target.path}",
+                because = "Summary tasks now registered from CiStep on root project, one per PlanSlug+JobSlug key, " +
+                    "to make cross-app dependency on single report possible"
             )
```

Comment:
> CONFIGURATION IS NOT BACKWARD COMPATIBLE

Proposal: keep empty stubs for old extensions to preserve compilation\configuration.
Ideally, we shouldn't check changes against the internal repository, but for now, it's risky to lose this feedback completely.

Reply:
Old extensions will work here for plugin, but not for CiStep

Reply:
I just want to preserve backward compatibility, to avoid breaking changes.
Don't know how to make it better and trying to speculate.

Reply:
Done, no more breaking changes